src/careamics_restoration/bioimage/io.py

```diff
@@ -0,0 +1,102 @@
+from pathlib import Path
+from typing import Union
+
+import yaml
+from bioimageio.core import load_resource_description
+from bioimageio.core.build_spec import build_model
+from bioimageio.spec.model.raw_nodes import Model
+
+from careamics_restoration.config.config import (
+    Configuration,
+)
+
+PYTORCH_STATE_DICT = "pytorch_state_dict"
+CAREAMICS_CONFIG = "careamics_config.yml"
+
+
+def _load_model_rdf(name: str) -> dict:
+    """Load an rdf(yaml) file given a model name."""
+    name = name.lower()
+    rdf_file = Path(__file__).parent.joinpath("rdfs").joinpath(name + ".yaml")
+    if not rdf_file.exists:
+        # a warning maybe?
+        return {}
+
+    with open(rdf_file) as f:
+        rdf = yaml.safe_load(f)
+
+    # add covers if available
+    cover_folder = Path(__file__).parent.joinpath("covers")
+    cover_images = list(cover_folder.glob(f"{name}_cover*.*"))
+    if len(cover_images) > 0:
+        rdf["covers"] = [str(img.absolute()) for img in cover_images]
+
+    # add documentation (md file)
+    doc = Path(__file__).parent.joinpath("docs").joinpath(f"{name}.md")
+    if doc.exists() > 0:
+        rdf["documentation"] = str(doc.absolute())
+
+    return rdf
+
+
+def get_default_model_specs(name: str) -> dict:
+    """Return the default specs for given model's name."""
+    return _load_model_rdf(name)
+
+
+def build_zip_model(
+    config: Configuration,
+    model_specs: dict,
+) -> Model:
+    """Build bioimage model zip file from model specification data.
+
+    Parameters
+    ----------
+        config (Configuration): Careamics' configuration,
+        model_specs (dict): Model's specs to export.
+
+    Return
+    ------
+        A bioimage raw Model
+    """
+    workdir = config.working_directory
+    # attach the best checkpoint to the bioimage model
+    checkpoint_path = workdir.joinpath(f"{config.experiment_name}_best.pth").absolute()
+    # build model zip
+    raw_model = build_model(
+        root=str(Path(model_specs["output_path"]).parent.absolute()),
+        attachments={"files": [str(checkpoint_path)]},
```

Comment:
This means that we save twice the weights or model state_dict. I don't have a good notion or how heavy they can be, but I am uneasy with this idea in general.

It would be better to save:
- configuration
- optimizer state_dict
- scheduler state_dict
- grad_scaler state_dict

as additional files, rather than the whole checkpoint.